Snowflake.Data/Client/SnowflakeDbCommand.cs

```diff
@@ -238,6 +238,15 @@ public override void Prepare()
             throw new NotImplementedException();
         }
 
+        public string GetQueryId()
+        {
+            SnowflakeDbConnection conn = (SnowflakeDbConnection)connection;
+            if ((conn != null) && (conn.SfSession != null))
+            {
+                return conn.SfSession.lastQueryId;

```

Comment:
Are you sure that this is the best way? What would happen in the following situation (pseudocode):
```
command1 = conn.createCommand
command2 = conn.createCommand
command1.execute
command2.execute
command1.GetQueryId() <---- here will be the query id of the second command?
```

Reply:
make sense. Will put queryId at statement level.

Reply:
done.